presto-lark-sheets/src/main/java/com/facebook/presto/lark/sheets/api/SheetsApiFactory.java

```diff
@@ -0,0 +1,249 @@
+                    throw new PrestoException(SheetsErrorCode.SHEET_BAD_DATA,
+                            format("Sheet %s.%s is empty", token, sheetId));
+                }
+                if (!(values[0] instanceof List)) {
+                    throw new PrestoException(SHEETS_API_ERROR,
+                            format("Illegal response data of sheet %s @ %s", token, range));
+                }
+                List<?> header = ((List<?>) values[0]);
+                return header.stream().map(obj -> obj == null ? null : obj.toString()).collect(Collectors.toList());
+            }
+            catch (Exception e) {
+                throw wrapApiError(e, format("Could not get data of sheet %s @ %s", token, range));
+            }
+        }
+
+        @Override
+        public SheetValues getValues(String token, String range)
+        {
+            try {
+                SheetsService.SpreadsheetsValuesGetReqCall request = sheetsService.getSpreadsheetss().valuesGet();
+                request.setSpreadsheetToken(token);
+                request.setRange(range);
+
+                Response<SpreadsheetsValuesGetResult> response = request.execute();
+                checkResponse(response);
+                SpreadsheetsValuesGetResult data = response.getData();
+
+                // Convert Object[] to List<List<Object>>
+                Object[] arr = data.getValueRange().getValues();
+                List<List<Object>> values = new ArrayList<>(arr.length);
+                for (Object item : arr) {
+                    if (item instanceof List) {
+                        values.add(((List) item));
+                    }
+                    else {
+                        throw new PrestoException(SHEETS_API_ERROR,
+                                format("Illegal response data of sheet %s @ %s", token, range));
+                    }
+                }
+
+                return new SheetValues(
+                        data.getRevision(),
+                        data.getValueRange().getRange(),
+                        values);
+            }
+            catch (Exception e) {
+                throw new PrestoException(SHEETS_API_ERROR, format("Could not get data of sheet %s @ %s", token, range), e);
+            }
+        }
+    }
+
+    private static void checkResponse(Response<?> response)
+    {
+        if (response.getCode() != 0) {
+            throw new PrestoException(SHEETS_API_ERROR,
+                    format("Bad response: [%d] %s", response.getCode(), response.getMsg()));
+        }
+    }
+
+    private static SheetInfo toMetaInfoSheet(String token, Sheet sheet)
+    {
+        return new SheetInfo(
+                token,
+                sheet.getSheetId(),
+                sheet.getTitle(),
+                sheet.getIndex(),
+                sheet.getColumnCount(),
+                sheet.getRowCount());
+    }
+
+    private static Domain toLarkDomain(SheetsConfig.Domain domain)
```

Comment:
Moved the utility function like `loadAppSecret` to SheetsUtil.

For `toLarkDomain`, it translates SheetsConfig.Domain to Lark SDK Domain. As we encapsulated all SDK invocations into `SheetsApiFactory`, it would be better to keep `toLarkDomain` remain in this class. What do you think?

Reply:
Sure, thanks for the clarification.